### ISSUE TITLE
Firebird 4 regularly crashes in Super Classic mode

### ISSUE THREAD

**Issue by grzegorzzyla** [FirebirdSQL/firebird#7495]
Attempting to run Firebird4 in SuperClassic mode always kills the server.
I have no idea how to track this, on a test machine that is cloned from the virtual one, there is no way to track which query kills the server.
There is no problem in Classic mode, more than 600-700 users are working (there is no information about killing the session in the logs).
I have 4 core files from today.

https://eclsoft.com/soft/uploads

SRV-01-SI-4-PRD	Mon Mar  6 04:42:01 2023
	Shutting down the server with 22 active connection(s) to 1 database(s), 0 active service(s)
SRV-01-SI-4-PRD	Mon Mar  6 04:42:01 2023
	INET/inet_error: read errno = 110, client host = it-06281, address = 10.1.153.86/63614, user = it-06281.zpue-w2
SRV-01-SI-4-PRD	Mon Mar  6 04:42:03 2023
	/usr/local/firebird/bin/fbguard: /usr/local/firebird/bin/firebird terminated
SRV-01-SI-4-PRD	Mon Mar  6 04:42:33 2023
	/usr/local/firebird/bin/fbguard: guardian starting /usr/local/firebird/bin/firebird

SRV-01-SI-4-PRD	Mon Mar  6 08:06:15 2023
	INET/inet_error: read errno = 104, client host = srv-01-trm1-s2, address = 10.1.1.64/58170, user = xxx.yyy
SRV-01-SI-4-PRD	Mon Mar  6 08:06:47 2023
	/usr/local/firebird/bin/fbguard: /usr/local/firebird/bin/firebird terminated abnormally (-1)
SRV-01-SI-4-PRD	Mon Mar  6 08:06:47 2023
	/usr/local/firebird/bin/fbguard: guardian starting /usr/local/firebird/bin/firebird

SRV-01-SI-4-PRD	Mon Mar  6 08:11:13 2023
	Shutting down the server with 171 active connection(s) to 2 database(s), 0 active service(s)
SRV-01-SI-4-PRD	Mon Mar  6 08:11:17 2023
	/usr/local/firebird/bin/fbguard: /usr/local/firebird/bin/firebird killed (did not terminate)
SRV-01-SI-4-PRD	Mon Mar  6 08:11:53 2023
	/usr/local/firebird/bin/fbguard: guardian starting /usr/local/firebird/bin/firebird
SRV-01-SI-4-PRD	Mon Mar  6 08:11:53 2023
	setsockopt: error setting IPV6_V6ONLY to 0
SRV-01-SI-4-PRD	Mon Mar  6 08:11:59 2023
	INET/inet_error: read errno = 104

**Comment by hvlad** [comment]
Is it reproducible ? If yes, and if you use 4.0.2 - could you try current snapshot build ?

**Comment by grzegorzzyla** [comment]
Yes, always in SuperClassic mode. Unfortunately I have my build that allows me to make metadata changes - if I need to restructure the tables. I can also merge the current state into my sources and build a version.

**Comment by grzegorzzyla** [comment]
Ok, I commented out this line, testing tonight.
For the record, I've included more detailed backtraces from the existing core files (core1-debug, core2-debug, etc.)
https://eclsoft.com/soft/uploads

**Comment by grzegorzzyla** [comment]
I compiled 4.0.2 with this fix, i.e. without the line port->port_context = nullptr;
In the morning there was a hang, the server dumped 2 core files. Unfortunately, trying to suspect gdb doesn't show anything sensible :(
In gdb I type thread apply all bt. Effects in core-debug.log
https://eclsoft.com/soft/uploads/08032023/

**Comment by hvlad** [comment]
I'm not a Linux expert, for me it looks like gdb requires root privs - try to run it with sudo

**Comment by hvlad** [comment]
Looks like minimal patch not helped, thus I recommend to merge with most current state of v4.0-release and try again.

**Comment by grzegorzzyla** [comment]
On this server i am running as root ;)
So I thought it would be better to merge from the current master.

**Comment by grzegorzzyla** [comment]
https://eclsoft.com/soft/uploads/11032023/
core1357.lz4 and core0736.lz4 with correspondent .log files is server crash dump.
core1603.lz4 core1659.lz4 is simply shutdown (service firebird stop).

OS - Centos 8 Stream.
Compilling in gcc-11.

**Comment by grzegorzzyla** [comment]
One more idea - I just disabled SELinux. I completely forgot about it. Maybe it matters for the SuperClassic.

**Comment by AlexPeshkoff** [comment]
First of all you should avoid the following lines in gdb:

**Comment by grzegorzzyla** [comment]
Yes, but how to do this?
[root@localhost 11032023]# file -Pelf_phnum=10000 core0736
core0736: ELF 64-bit LSB core file, x86-64, version 1 (SYSV), SVR4-style, from '/usr/local/firebird/bin/firebird', real uid: 84, effective uid: 84, real gid: 84, effective gid: 84

core seems to be dumped by firebird.
https://eclsoft.com/soft/uploads/11032023/core0736.log

**Comment by grzegorzzyla** [comment]
OK, new cores for analysis.

https://eclsoft.com/soft/uploads/20032023/core0809.log
https://eclsoft.com/soft/uploads/20032023/core0750.log
